### PR TITLE
Use `kcopy!` and `fill!` instead of broadcasting in `cg`

### PR DESCRIPTION
**Pull request by glwagner**

### CODE REVIEW

File: src/cg.jl [hunk 1]
```diff
@@ -182,7 +182,7 @@ kwargs_cg = (:M, :ldiv, :radius, :linesearch, :atol, :rtol, :itmax, :timemax, :v
           inconsistent = !linesearch
         end
         if linesearch
-          iter == 0 && (x .= b)
+          iter == 0 && (kcopy!(x, b))
```
Comment (glwagner) [inline src/cg.jl:185]:
```suggestion
          iter == 0 && (@kcopy!(x, b))
```

File: src/cg.jl [hunk 2]
```diff
@@ -133,15 +133,15 @@ kwargs_cg = (:M, :ldiv, :radius, :linesearch, :atol, :rtol, :itmax, :timemax, :v
...
     z = MisI ? r : solver.z
 
-    x .= zero(FC)
+    fill!(x, zero(FC))
     if warm_start
       mul!(r, A, Δx)
       @kaxpby!(n, one(FC), b, -one(FC), r)
     else
-      r .= b
+      @kcopy!(r, b)
```
Comment (glwagner) [inline src/cg.jl:141]:
```suggestion
      @kcopy!(n, r, b)
```

File: src/cg.jl [hunk 3]
```diff
@@ -133,15 +133,15 @@ kwargs_cg = (:M, :ldiv, :radius, :linesearch, :atol, :rtol, :itmax, :timemax, :v
...
     end
     MisI || mulorldiv!(z, M, r, ldiv)
-    p .= z
+    @kcopy!(p, z)
```
Comment (glwagner) [inline src/cg.jl:144]:
```suggestion
    @kcopy!(n, p, z)
```

**Comment by amontoison** [comment]
Done in #899 and #897

**Pull request closed without merging**